packages/app-builder-lib/src/node-module-collector/nodeModulesCollector.ts

```diff
@@ -48,70 +90,77 @@ export abstract class NodeModulesCollector {
     }
   }
 
-  private convertToDependencyGraph(tree: DependencyTree): DependencyGraph {
-    const result: DependencyGraph = { ".": {} }
-
-    const flatten = (node: DependencyTree, parentKey = ".") => {
-      const dependencies = node.dependencies || {}
-
-      for (const [key, value] of Object.entries(dependencies)) {
-        // Skip empty dependencies(like some optionalDependencies)
-        if (Object.keys(value).length === 0) {
-          continue
-        }
-        const version = value.version || ""
-        const newKey = `${key}@${version}`
-        this.dependencyPathMap.set(newKey, path.normalize(this.resolvePath(value.path)))
-        if (!result[parentKey]?.dependencies) {
-          result[parentKey] = { dependencies: [] }
-        }
-        result[parentKey].dependencies!.push(newKey)
-
-        if (node.__circularDependencyDetected) {
-          continue
-        }
-        flatten(value, newKey)
+  private convertToDependencyGraph(tree: DependencyTree, parentKey = "."): DependencyGraph {
+    return Object.entries(tree.dependencies || {}).reduce<DependencyGraph>((acc, curr) => {
+      const [packageName, dependencies] = curr
+      // Skip empty dependencies (like some optionalDependencies)
+      if (Object.keys(dependencies).length === 0) {
+        return acc
+      }
+      const version = dependencies.version || ""
+      const newKey = `${packageName}@${version}`
+      // Map dependency details: name, version and path to the dependency tree
+      this.dependencyPathMap.set(newKey, path.normalize(this.resolvePath(dependencies.path)))
```

Comment:
If the `path` is empty or `undefined`, could we add a debug log here? Several bugs that have been discovered recently often occur when the `path` is empty.

Reply:
Definitely can do